servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DefaultDnsClient.java

```diff
@@ -968,6 +970,14 @@ private static <T, A> Publisher<? extends Collection<ServiceDiscovererEvent<T>>>
         });
     }
 
+    private static boolean shouldRevokeState(final Throwable t) {
+        // ISE => Subscriber exceptions (downstream of retry)
+        return t instanceof SrvAddressRemovedException || t instanceof IllegalStateException ||
+                t instanceof ClosedDnsServiceDiscovererException || (NX_DOMAIN_INVALIDATES &&
+                t.getCause() instanceof DnsErrorCauseException &&
+                NXDOMAIN.equals(((DnsErrorCauseException) t.getCause()).getCode()));
```

Comment:
By creating a hard dependency on a new netty type `DnsErrorCauseException` we limit users ability to downgrade Netty version independently from ST. It usually creates a lot of classpath pain, especially if users classpath is messy. Consider either checking the type by its string name instead of `instanceof` or use a `MethodHandler` to check if `DnsErrorCauseException` type is available at runtime as part of initializing `NX_DOMAIN_INVALIDATES` value.